lib/rules/spaced-comment.js

```diff
@@ -51,37 +51,46 @@ function parseMarkersOption(markers) {
 
 /**
  * Creates RegExp object for `always` mode.
- * Generated pattern is below:
+ * Generated pattern for beginning of comment:
  *
  * 1. First, a marker or nothing.
  * 2. Next, a space or an exception pattern sequence.
  *
+ * Generated pattern for end of comment:
+ *
+ * 1. A space or exception pattern sequence.
+ * 2. End of comment.
+ *
  * @param {string[]} markers - A marker list.
  * @param {string[]} exceptions - A exception pattern list.
- * @returns {RegExp} A RegExp object for `always` mode.
+ * @param {string} location - Specifies if pattern checks the beginning or end of a comment
+ * @returns {RegExp} A RegExp object for the beginning of a comment in `always` mode.
  */
-function createAlwaysStylePattern(markers, exceptions) {
-    var pattern = "^";
+function createAlwaysStylePattern(markers, exceptions, location) {
```

Comment:
You can also reduce the complexity of this function by extracting the "exception generation" bits into a function, and calling that instead of passing in "begin" or "end", because the "end" code path basically uses none of the body of this code except for exception generation.